### PR TITLE
Fix quest saving #247

### PR DESCRIPTION
**Pull request by AleksandrShvartz**

### CODE REVIEW

File: website/questmaker/tests/excursion.sql [hunk 1]
```diff
@@ -13,7 +13,7 @@ INSERT INTO quests (title, author_id, description, keyword, password, time_open,
 
 --places info
 --branch "Temple architecture"
-INSERT INTO places (coords, radius) VALUES ('(59.8633917835336, 30.472641018135985)', 50);
+INSERT INTO places (coords, radius) VALUES ( '(59.8633917835336, 30.472641018135985)', 50);
```
Comment (aVorotnikov) [inline website/questmaker/tests/excursion.sql:16]: For what? Why?
```suggestion
INSERT INTO places (coords, radius) VALUES ('(59.8633917835336, 30.472641018135985)', 50);
```

Comment (IlyaP01) [reply website/questmaker/tests/excursion.sql:16]: image
For what?

File: website/questmaker/api/constructor.py [hunk 2]
```diff
@@ -1,12 +1,13 @@
 from questmaker.quest import Quest, Question, Place, Hint, Answer, Movement, File, update_from_dict
 from questmaker.quest_container import EntityType, QuestContainer
-from questmaker.db import get_draft, update_draft, write_draft, remove_draft, get_draft_for_update
+from questmaker.db import get_draft, update_draft, write_draft, remove_draft, get_draft_for_update, get_db
```
Comment (IlyaP01) [inline website/questmaker/api/constructor.py:3]: Remove extra import

File: website/questmaker/db.py [hunk 3]
```diff
@@ -440,33 +440,22 @@ def get_question_types():
...
-                cursor.execute('INSERT INTO files (url_for_file,f_type_id) '
-                               'VALUES (%s,%s) RETURNING f_id', (file.url, type_id))
-                return cursor.fetchone()['f_id']
-            else:
-                return False
-    else:
-        ids = []
+    :return: True if success, False if not
+    """
+    with get_db(), get_db().cursor(cursor_factory=psycopg2.extras.DictCursor) as cursor:
```
Comment (IlyaP01) [inline website/questmaker/db.py:449]: Remove autocommit

File: website/questmaker/db.py [hunk 4]
```diff
@@ -440,33 +440,22 @@ def get_question_types():
...
         cursor.execute('SELECT f_type_id FROM file_types WHERE f_type_name = %s', (file.type,))
         type_id = cursor.fetchone()['f_type_id']
         if not (type_id is None):
```
Comment (IlyaP01) [inline website/questmaker/db.py:452]: It's possible to write "type_id is not None"

File: website/questmaker/quest.py [hunk 5]
```diff
@@ -406,6 +421,16 @@ def to_dict(self):
                 'pos_x': self.pos_x,
                 'pos_y': self.pos_y}
 
+    def to_db(self):
```
Comment (IlyaP01) [inline website/questmaker/quest.py:424]: This function doesn't write question to db. Rename it or refactor, e.g.  add set_question inside this function.

File: website/questmaker/db.py [hunk 6]
```diff
@@ -506,18 +495,20 @@ def set_quest(quest):
...
-                       'VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s) RETURNING quest_id',
-                       (quest.title, quest.author_id, quest.description,
+        cursor.execute('INSERT INTO quests (title, author_id, description, keyword, password, '
+                       'time_open, time_close, lead_time, cover_url, hidden, published) '
+                       'VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s) RETURNING quest_id',
+                       (quest.title, quest.author_id, quest.description, quest.keyword,
                         quest.password, quest.time_open, quest.time_close, quest.lead_time,
-                        quest.cover_url, str(quest.hidden).upper()))
-        return cursor.fetchone()['quest_id']
+                        quest.cover_url, str(quest.hidden).upper(), quest.published))
```
Comment (IlyaP01) [inline website/questmaker/db.py:509]: `str(quest.hidden).upper()`?

**Review by IlyaP01 — CHANGES_REQUESTED** [review]

**Review by IlyaP01 — APPROVED** [review]

**Review by BaevDaniil — APPROVED** [review]

**Pull request merged**